skorch/tests/test_net.py

```diff
@@ -858,23 +858,23 @@ def test_setting_lr_after_init_reflected_in_optimizer(
 
     @pytest.mark.parametrize('kwargs,expected', [
         ({}, ""),
+        (
+            # virtual params should prevent re-initialization
+            {'optimizer__lr': 0.12, 'optimizer__momentum': 0.34},
+            ("")
+        ),
         (
             {'module__input_units': 12, 'module__hidden_units': 34},
             ("Re-initializing module because the following "
              "parameters were re-set: hidden_units, input_units.\n"
              "Re-initializing optimizer.")
         ),
         (
-            {'optimizer__lr': 0.12, 'optimizer__momentum': 0.34},
-            ("Re-initializing optimizer because the following "
-             "parameters were re-set: lr, momentum.")
-        ),
-        (
-            {'module__input_units': 12, 'module__hidden_units': 34, 'lr': 0.56},
+            {'module__input_units': 12, 'module__hidden_units': 34,
+             'optimizer__momentum': 0.56},
             ("Re-initializing module because the following "
              "parameters were re-set: hidden_units, input_units.\n"
-             "Re-initializing optimizer because the following "
-             "parameters were re-set: lr.")
+             "Re-initializing optimizer.")
```

Comment:
I would propose changing the error message to `"Re-initializing optimizer (triggered by another component)."` in this case. This should be straightforward to add [here](https://github.com/dnouri/skorch/blob/1700cbb22e2c49b1a6431a27d4f17e3bdfe54b09/skorch/net.py#L453).

Reply:
But that would be the default then. Whenever the optimizer is re-initialized it would say "by another component", even if the user calls `initialize_optimizer` directly. Do we want that? It's technically correct but are there cases where this might be confusing? I couldn't find any but I thought I bring this up first.

Reply:
Since `triggered_directly=True` is the default, the proposed message would not trigger by default but only when `initialize_optimizer` is triggered indirectly, so this should be good or am I missing something?

That said, when the user calls `initialize_optimizer` directly, it will trigger the other case ("...because the following parameters were re-set: ..."), which may be confusing?